composer/conftest.py

```diff
@@ -21,5 +24,16 @@
 @pytest.fixture(scope="session")
 def airflow_database():
     import airflow.utils.db
+
+    # We use separate directory for local db path per session.
+    tmp_dir = tempfile.TemporaryDirectory()
+    os.environ['AIRFLOW_HOME']  = tmp_dir.name
```

Comment:
I'm not 100% sure that this will work. It's possible the Airflow config has already been parsed by the time we get here.

https://github.com/apache/airflow/blob/1cdd1cd3a356e95975357da2dc65ff2afb2a63c7/airflow/utils/db.py#L701

If you leave off the cleanup step, can you confirm that a file is created at `$AIRFLOW_HOME/airflow.db` ? This is where I'd expect it per the default config:

https://github.com/apache/airflow/blob/86695b62a0281364088642fa3dc17d92cf9e7cbe/airflow/config_templates/default_airflow.cfg#L59

Reply:
If not, something else we can try is to set `airflow.settings.SQL_ALCHEMY_CONN`

https://github.com/apache/airflow/blob/1.10.14/airflow/settings.py#L146

This variable is used to connect to the database from here: https://github.com/apache/airflow/blob/1.10.14/airflow/utils/db.py#L385

Reply:
Unfortunately you're right. I managed to get it to work by setting it before running pytest.